config/config.go

```diff
@@ -545,6 +545,17 @@ func ParseConfigFile(filename string, terragruntOptions *options.TerragruntOptio
 		return nil, err
 	}
 
+	// If IamRole wasn't provided, and if it is config file, try first to decode IamRole
+	if include == nil && terragruntOptions.IamRole == "" {
+		iamConfig, err := PartialParseConfigString(configString, terragruntOptions, include, filename, []PartialDecodeSectionType{TerragruntFlags})
```

Comment:
What happens if the `iam_role` param depends on a `local` var? Or a `dependency` block? Or anything else that hasn't been parsed yet? In other words, do we need to update the docs on parse order to call out any new limitations here?